File: go/162.go

package main
//
//import "fmt"

// 1 star
// slow, beat 2%, 可以改为使用二分法
func findPeakElement(nums []int) int {
	length := len(nums)
	if length == 1 {return 0}
	for index, _ := range nums {
		if index == 0 {
			if nums[index] > nums[index+1] {
				return index}
		} else if index == length-1 {
			if nums[index] > nums[index-1] {
				return index}
		} else {
			if nums[index] > nums[index-1] && nums[index] > nums[index+1] {
				return index
			}
		}
	}
	return 0
}


//func main() {
//	nums := []int {1}
//	fmt.Println(findPeakElement(nums))
//}

File: v2/162.py



class Solution:
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # 2 star, 直接从头到尾遍历
        length = len(nums)
        if length == 1:
            return 0
        for i in range(length):
            if i == 0:
                if nums[i] > nums[i+1]:
                    return i
            elif i == length - 1:
                if nums[i] > nums[i-1]:
                    return i
            else:
                if nums[i] > nums[i+1] and nums[i] > nums[i-1]:
                    return i



